FileIO/AsciiRasterInterface.cpp

```diff
@@ -215,4 +216,29 @@ void AsciiRasterInterface::writeRasterAsASC(GeoLib::Raster const& raster, std::s
     out.close();
 }
 
-} // end namespace GeoLib
+
+/// Checks if all raster files actually exist
+static bool allRastersExist(std::vector<std::string> const& raster_paths)
+{
+	for (auto raster = raster_paths.begin(); raster != raster_paths.end();
+	     ++raster)
+	{
+		std::ifstream file_stream(*raster, std::ifstream::in);
+		if (!file_stream.good()) return false;
+		file_stream.close();
+	}
+	return true;
+}
+
+boost::optional<std::vector<GeoLib::Raster const*>> readRasters(
+    std::vector<std::string> const& raster_paths)
+{
+	if (!allRastersExist(raster_paths)) return boost::none;
```

Comment:
Only a note: Actually it is dangerous to first check if files exist, then believe that they exist and just try to read them blindly. The files could have vanished in between. @endJunction, you shouldn't spend time on that now, especially since the problem has not been introduced in this PR. But for the future we should keep in mind that files should be read in one go and not be checked for existence prior to opening.